mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -6753,10 +6756,111 @@ struct ThreadwiseCopyRewritePattern
         SmallVector<Value, 8> destLowerIndices =
             layeredDestIndices[layeredDestIndices.size() - 1];
 
+        bool toEmitOOBStoreCheckLogic = false;
+        SmallVector<unsigned, 2> oobStoreCheckDims;
+        if (composedDestTransform && boundCheckDestAttr) {
+          if (boundCheckDestAttr.size() ==
+              composedDestTransform.getNumResults()) {
+            for (unsigned iter = 0; iter < boundCheckDestAttr.size(); ++iter) {
+              if (boundCheckDestAttr[iter]
+                      .template cast<IntegerAttr>()
+                      .getInt()) {
+                toEmitOOBStoreCheckLogic = true;
+                oobStoreCheckDims.push_back(iter);
+              }
+            }
+          }
+        }
+
         // Store to dest.
-        // Issue scalar store.
-        b.create<StoreOp>(loc, convertedScalarValue, op.dest(),
-                          destLowerIndices);
+        // Issue scalar store ,oob store only support f32 now
+        if (toEmitOOBStoreCheckLogic &&
+            destType.getElementType() == b.getF32Type()) {
+          SmallVector<Value, 8> destLowerStoreIndices;
+          SmallVector<Value, 8> destLowerStoreOOBIndices;
+
+          for (int i = 0; i < destLowerIndices.size(); i++) {
```

Comment:
nit picking: use `unsigned` or `size_t` to remove compile time warnings. also it's a good practice in C++ to use `++i` instead of `i++`.